src/CoreWCF.Primitives/src/CoreWCF/Channels/SslStreamSecurityUpgradeProvider.cs

```diff
@@ -314,7 +314,7 @@ protected override async Task<(Stream, SecurityMessageProperty)> OnAcceptUpgrade
             return true;
         }
 
-        public override SecurityMessageProperty GetRemoteSecurity()
+        public override async Task<SecurityMessageProperty> GetRemoteSecurityAsync()
         {
```

Comment:
I did a quick deep dive into the implementation and usage here and this method doesn't need to be async. Other than the fact that an X509Authenticator actually has a sync implementation, even if it was async, this is duplicating effort from ValidateRemoteCertificate. The changes needed are to pass `true` to the second parameter of X509SecurityToken constructor in ValidateRemoteCertificate, and in the `if (_clientCertificate!=null)` code block here return the existing _clientSecurity that was created during ValidateRemoteCertificate.